src/python/py/models/builder.py

```diff
@@ -108,8 +116,58 @@ def __init__(self, config, io_dtype, onnx_dtype, ep, cache_dir, extra_options):
             }
         }
 
-    def save(self, out_path):
-        print("Saving ONNX model")
+    def make_genai_config(self, model_name_or_path, extra_kwargs, out_dir):
+        config = GenerationConfig.from_pretrained(model_name_or_path, **extra_kwargs)
+        hf_default_search_values = {
+            "max_length": 20,
+            "pad_token_id": None,
+            "top_k": 50,
+        }
+
+        genai_config = {
+            "model": {
+                "bos_token_id": config.bos_token_id,
+                "context_length": self.context_length,
+                "decoder": self.filename,
+                "eos_token_id": config.eos_token_id,
+                "head_size": self.head_size,
+                "hidden_size": self.hidden_size,
+                "logits_type": "float32" if self.io_dtype == TensorProto.FLOAT else "float16",
+                "kv_type": "float32" if self.io_dtype == TensorProto.FLOAT else "float16",
+                "num_attention_heads": self.num_attn_heads,
+                "num_hidden_layers": self.num_layers,
+                "num_key_value_heads": self.num_kv_heads,
+                "pad_token_id": config.pad_token_id if hasattr(config, "pad_token_id") and config.pad_token_id != hf_default_search_values["pad_token_id"] else config.eos_token_id,
+                "type": self.model_type[ : self.model_type.find("For")].lower(),
+                "vocab_size": self.vocab_size,
+            },
+            "search": {
+                "diversity_penalty": config.diversity_penalty if hasattr(config, "diversity_penalty") else 0.0,
+                "length_penalty": config.length_penalty if hasattr(config, "length_penalty") else 1.0,
+                "max_length": config.max_length if hasattr(config, "max_length") and config.max_length != hf_default_search_values["max_length"] else self.context_length,
+                "min_length": 0,
+                "no_repeat_ngram_size": config.no_repeat_ngram_size if hasattr(config, "no_repeat_ngram_size") else 0,
+                "num_beams": config.num_beams if hasattr(config, "num_beams") else 1,
+                "num_return_sequences": config.num_return_sequences if hasattr(config, "num_return_sequences") else 1,
+                "repetition_penalty": config.repetition_penalty if hasattr(config, "repetition_penalty") else 1.0,
+                "temperature": config.temperature if hasattr(config, "temperature") else 1.0,
+                "top_k": config.top_k if hasattr(config, "top_k") and config.top_k != hf_default_search_values["top_k"] else 0,
```

Comment:
why do we need to check "config.top_k != hf_default_search_values["top_k"]"? 
With the check, top_k will be set to 0 if the top_k in config is same as hf_default_search_values["top_k"]

Reply:
If the top k is specified by the HuggingFace config, we want to copy that value into our config.
If the top k is not specified by the HuggingFace config, we want to set to a default value of 50

Reply:
Re: max length:

If the maximum length is specified by the HuggingFace config, we want to copy that value into our config
If the maximum length is not specified by the HuggingFace config, we want to set to a default value of the model max length